mri/operators/proximity/weighted.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+
+from modopt.opt.proximity import SparseThreshold
+
+
+class WeightedSparseThreshold(SparseThreshold):
+    """This is a weighted version of `SparseThreshold` in ModOpt.
+    When chosen `scale_based`, this applied weighted proximity as :
+        W * P^i where i is the scale
```

Comment:
Not clear here what `W` and `P^i` are

Reply:
I think I know understand given what's underneath, but I don't understand where this rule comes from, do you have a reference?

Reply:
Ill try to add it.

Reply:
@philouc can you please help me find a way to get the reference for variable scale? I just kept it as generic as possible, but I know for a fact that 1  / \lamda ^ s where s is scale is used constantly..
Here we can set any base scale and any power (as a geometric progression)

Reply:
Hey Chaithya, I have to find it in Pesquet & Chouzenoux papers as in Loubna 2019 ISBI's we did not cite the exact reference even though we wrote the right formulae in Eq. (4)